Core/src/org/sleuthkit/autopsy/coreutils/ExecUtil.java

```diff
@@ -213,6 +214,40 @@ public static void killProcess(Process process) {
     }
 
     /**
+     * Shuts down a task executor service, waiting until all tasks are
+     * terminated. The current policy is to wait for the tasks to finish so that
+     * the case for which the executor is running can be left in a consistent
+     * state.
+     *
+     * @param executor The executor.
+     */
+    public static void shutDownTaskExecutor(ExecutorService executor) {
+        executor.shutdown();
+        boolean taskCompleted = false;
+        while (!taskCompleted) {
+            try {
+                taskCompleted = executor.awaitTermination(DEFAULT_TIMEOUT, DEFAULT_TIMEOUT_UNITS);
+            } catch (InterruptedException ignored) {
+                /*
+                 * The current policy is to wait for the task to finish so that
+                 * the case can be left in a consistent state.
+                 *
+                 * For a specific example of the motivation for this policy,
+                 * note that a application service (Solr search service)
+                 * experienced an error condition when opening case resources
+                 * that left the service blocked uninterruptibly on a socket
+                 * read. This eventually led to a mysterious "freeze" as the
+                 * user-cancelled service task continued to run holdiong a lock
+                 * that a UI thread soon tried to acquire. Thus it has been
+                 * deemed better to make the "freeze" happen in a more
+                 * informative way, i.e., with the progress indicator for the
+                 * unfinished task on the screen, if a similar error condition
+                 * arises again.
+                 */
```

Comment:
This comment seems pretty specific to the Case class use case. now that we are "going public", I think it would be a good idea to remove this comment and change the method documentation to leave it at "Shuts down a task executor service, waiting until all tasks are terminated."